src/components/common/SearchBar.js

```diff
@@ -0,0 +1,70 @@
+import React, { Component } from "react";
+import { Text, View, ScrollView, Dimensions } from "react-native";
+import { SearchInput, SearchResults } from ".";
+import { APIConstants } from "../../constants/APIConstants";
+import PortFolioConstants from "../../constants/PortfolioConstants";
+import CreateStockEntry from "../manage/stock/CreateStockEntry";
+
+class SearchBar extends Component {
+  constructor(props) {
+    super(props);
+    this.onSelection = this.onSelection.bind(this);
```

Comment:
Avoid using `fn.bind` it is an anti-pattern. Use arrow operators instead. For instance remove this assignment. and in the template do:
```
<SearchResults
     onSelection={() => this.onSelection()}
     searchResults={this.state.searchResults}
/>
```

Another way to do this would be to make this function an instance function by doing:
```
   onSelection = (companyInfo) => {
    let searchInput = "";
    let searchResults = [];
    let lastSelection = companyInfo;
    this.setState({ searchInput, searchResults, lastSelection });
  }
```

But the former approach is the standard way:

Reply:
<SearchResults
            onSelection={() => this.onSelection}
            searchResults={this.state.searchResults}
          />
 doesn't work. It works only for direct setting of state within the arrow function.

Reply:
If you did the way you mentioned in the comment it won't work. You need to invoke the function inside arriw function. Try my example and see if that works.